internal/connector/internal/handlers/connectors.go

```diff
@@ -389,6 +389,29 @@ func validateConnectorPatch(bytes []byte, ct *dbapi.ConnectorType) *errors.Servi
 	return nil
 }
 
+func validateConnectorImmutableProperties(patch public.ConnectorRequest, originalResource public.Connector) handlers.Validate {
+	// Check User is not attempting to patch an immutable property
+	return func() *errors.ServiceError {
+		valid := true
+		if patch.ConnectorTypeId != originalResource.ConnectorTypeId {
+			valid = false
+		}
+		if patch.NamespaceId != originalResource.NamespaceId {
+			valid = false
+		}
+		if patch.Channel != originalResource.Channel {
+			valid = false
+		}
+
+		if !valid {
+			err := errors.BadRequest("An attempt was made to modify an immutable field.")
```

Comment:
Can you include a hint about the offending field? That way the user recognize the source of the conflict and can fix the problem.

Reply:
Done.